mgmt/rest/client/plugin.go

```diff
@@ -78,6 +79,63 @@ func (c *Client) UnloadPlugin(pluginType, name string, version int) *UnloadPlugi
 	return r
 }
 
+// SwapPlugin swaps two plugins with the same type and name e.g. collector:mock:1 with collector:mock:2
+func (c *Client) SwapPlugin(loadPath []string, unloadType, unloadName string, unloadVersion int) *SwapPluginsResult {
+	r := &SwapPluginsResult{}
+
+	if len(loadPath) != 1 {
+		r.Err = errors.New("Did not receive only one plugin to load.")
+		return r
+	}
+	// Check if plugin you are trying to unload is loaded
+	var isLoaded bool
+	lps := c.GetPlugins(false)
```

Comment:
Client needs to be expanded to use GetPlugin from the server. This would return a single plugin or no plugin, and no need to iterate through all plugins in the system.